protocol/nats_jetstream/v3/protocol.go

```diff
@@ -0,0 +1,290 @@
+/*
+ Copyright 2024 The CloudEvents Authors
+ SPDX-License-Identifier: Apache-2.0
+*/
+
+package nats_jetstream
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+
+	"github.com/nats-io/nats.go"
+	"github.com/nats-io/nats.go/jetstream"
+)
+
+// Protocol is a reference implementation for using the CloudEvents binding
+// integration. Protocol acts as both a NATS client and a NATS handler.
+type Protocol struct {
+
+	// nats connection
+	conn      *nats.Conn
+	url       string
+	natsOpts  []nats.Option
+	connOwned bool
+
+	// jetstream options
+	jetSteamOpts []jetstream.JetStreamOpt
```

Comment:
is this used somewhere?

Reply:
`func New(ctx context.Context, opts ...ProtocolOption) (*Protocol, error) `
calls
`if p.jetStream, errConnection = jetstream.New(p.conn, p.jetSteamOpts...); errConnection != nil {`